src/main/java/org/jenkinsci/plugins/aquadockerscannerbuildstep/AquaDockerScannerBuilder.java

```diff
@@ -399,7 +410,13 @@ public boolean configure(StaplerRequest req, JSONObject formData) throws FormExc
 					password = Secret.fromString(authForm.getString("password"));
 					token = Secret.fromString("");
 				}
-				
+				if (registryAuthType.equals("pipelineAuth")) {
+					registryUsername = Secret.fromString(registryAuthForm.getString("registryUsername"));
+					registryPassword = Secret.fromString(registryAuthForm.getString("registryPassword"));
+				} else {
+					registryUsername = null;
+					registryPassword = null;
```

Comment:
Check whether setting null value throws any nil pointers or not

Reply:
@anupamtamrakar , I checked with null behaviour, no null pointer exceptions happening with this.

also tried with below scratch code to ensure it works.
`listener.getLogger().println("secret from null testing : " + Secret.toString(null));`
output: `secret from null testing : `